test/Makefile

```diff
@@ -12,10 +12,33 @@ $(TDIRS):
 clean:
 	-rm -fr out
 
-update:
+update: out/cmpll/ll_curr.txt out/cmpll/ll_prev.txt
+	comm -13 out/cmpll/ll_curr.txt out/cmpll/ll_prev.txt | xargs rm -fv
+	rm out/cmpll/ll_curr.txt out/cmpll/ll_prev.txt
 	for f in out/*/ll/*.ll; do \
 		ll=`echo $$f | sed -e 's;/ll/;/;' -e 's;^out;ll;'`; \
 		cmp -s $$f $$ll || (mkdir -p `dirname $$ll` && cp -p $$f $$ll); \
 	done
 
-.PHONY: $(TARGETS) $(TDIRS) clean update
+diff: out/cmpll/ll_curr.txt out/cmpll/ll_prev.txt
+	@diff out/cmpll/ll_curr.txt out/cmpll/ll_prev.txt | sed -e 's;^< ll/\(.*\)/;< out/\1/ll/;' -e '/^[<>]/!d'
+	@rm out/cmpll/ll_curr.txt out/cmpll/ll_prev.txt
+	@for curr in out/*/ll/*.ll; do \
+		( \
+			d=`echo $$curr | sed -e 's;\.ll$$;.diff;' -e 's;/ll/;/;' -e 's;^out;out/cmpll;'`; \
+			mkdir -p `dirname $$d`; \
+			prev=`echo $$curr | sed -e 's;/ll/;/;' -e 's;^out;ll;'`; \
+			[ -e $$prev ] && diff -u $$curr $$prev> $$d; \
+			[ -s $$d ] && echo $$d || rm -f $$d \
+	    )& \
```

Comment:
What is the `&` for? Doesn't that run it in the background? I can't see how that works.

Reply:
Yes. It was was taking couple of seconds to run, this speeds it up.

Reply:
Because now it runs in parallel.

Reply:
That's not the right way to do things.  You can't have background jobs in a Makefile!

Reply:
This is just far too complicated, is doing unnecessary work and so it is slow. All it needs to do is give a list of the files that are changed (files that were present before and after, but are different).  It should be doable with a find calling cmp.

Reply:
It does not need to diff them. It does not to list files that were added or removed.

Reply:
In the last meeting I mentioned that I am going to write diffs so one can click and view it, I thought you were OK with it. Let me change to just cmp.